packages/serialport/lib/serialport.ts

```diff
@@ -1,11 +1,11 @@
-import { ErrorCallback, OpenOptions, SerialPortStream } from '@serialport/stream'
-import { autoDetect, AutoDetectTypes } from '@serialport/bindings-cpp'
+import { ErrorCallback, OpenOptions, SerialPortStream, StreamOptions } from '@serialport/stream'
+import { autoDetect, AutoDetectTypes, OpenOptionsFromBinding } from '@serialport/bindings-cpp'
 
 const DetectedBinding = autoDetect()
```

Comment:
Since `autoDetect` returns a non-discriminating union (`AutoDetectTypes`), I think that is why the type assertion is needed from `DetectedBinding` later. Is it possible to [brand](https://medium.com/@KevinBGreene/surviving-the-typescript-ecosystem-branding-and-type-tagging-6cf6e516523d) each binding interface to help with this detection narrowing?